### PR TITLE
NEW: add boolean action to update status to success/failed

### PR DESCRIPTION
**Pull request by rodgomes**

### CODE REVIEW

File: tests/integration/test_scmsteprun_view.py [hunk 1]
```diff
@@ -158,3 +158,51 @@ def test_create(self, client, logged_in_user, scm_pipeline_run, scm_step_run):
         assert response.status_code == 201
         assert models.SCMStepRun.objects.filter(name='Release product').exists()
         assert models.SCMStepRun.objects.count() == initial_count + 1
+
+    def test_status_update_with_boolean(self, client, logged_in_user, scm_step_run):
+        assert scm_step_run.status == 'not started'
+
+        url = f'/scm-step-runs/{scm_step_run.public_identifier}/status_success/'
```
Comment (mattiassluis) [inline tests/integration/test_scmsteprun_view.py]: Should the URL not be `../status-success/`?

File: tests/integration/test_scmsteprun_view.py [hunk 2]
```diff
@@ -158,3 +158,51 @@ def test_create(self, client, logged_in_user, scm_pipeline_run, scm_step_run):
...
+        response = client.patch(url, data, content_type='application/json')
+        assert response.status_code == 204
+        p = models.SCMStepRun.objects.get(pk=scm_step_run.public_identifier)
+        assert p.status == 'success'
+
+    def test_status_update_with_string(self, client, logged_in_user, scm_step_run):
+        assert scm_step_run.status == 'not started'
+
+        url = f'/scm-step-runs/{scm_step_run.public_identifier}/status_success/'
+        data = {'status_success': 'yes'}
```
Comment (mattiassluis) [inline tests/integration/test_scmsteprun_view.py]: (Why) should this work? Should the request with JSON not only allow Boolean values?

File: katka/views.py [hunk 3]
```diff
@@ -97,6 +101,16 @@ def get_queryset(self):
         user_groups = self.request.user.groups.all()
         return super().get_queryset().filter(scm_pipeline_run__application__project__team__group__in=user_groups)
 
+    @action(detail=True, methods=['patch'])
+    def status_success(self, request, pk=None):
+        serializer = SCMStepRunStatusUpdateSerializer(data=request.data, context={'request': request})
+        serializer.is_valid(raise_exception=True)
+        pipeline_run = self.get_queryset().filter(pk=pk).first()
```
[the remark this answers is not in the document]
Comment (rodgomes) [reply katka/views.py]: ah, good point. That works. I thought it was magic that would only work when using the default dispatches. 

Regarding the order, for me is the same. Just that we can save a database call if the field is not valid, but i can change to be more consistent.

File: katka/views.py [hunk 4]
```diff
@@ -97,6 +101,16 @@ def get_queryset(self):
...
+        if not pipeline_run:
+            return Response(status=status.HTTP_404_NOT_FOUND)
+        serializer.update(pipeline_run, serializer.validated_data)
+        return Response(status=status.HTTP_204_NO_CONTENT)
```
[the remark this answers is not in the document]
Comment (rodgomes) [reply katka/views.py]: But this is already a custom action, I don;t see the need to return the object. You either change the status or not and the result is either success or not, so any status 200 would be sufficient and as we don't return anything thus 204

File: tests/integration/test_scmsteprun_view.py (hunk 2 above)
Comment (rodgomes) [reply tests/integration/test_scmsteprun_view.py]: that's the default behaviour of DRF boolean field.

File: tests/integration/test_scmsteprun_view.py (hunk 1 above)
Comment (rodgomes) [reply tests/integration/test_scmsteprun_view.py]: updated

File: tests/integration/test_scmsteprun_view.py (hunk 2 above)
Comment (mattiassluis) [reply tests/integration/test_scmsteprun_view.py]: Oh wow.. I checked the docs of both DRF and Django and did not see any mention of this but if it is default behaviour let's keep it that way

Comment (rodgomes) [reply tests/integration/test_scmsteprun_view.py]: yeah, i saw in the source code:
https://github.com/encode/django-rest-framework/blob/master/rest_framework/fields.py#L668
Just added the test to make visible that works like that

**Pull request closed without merging**